src/ConnectedMode/BranchMatcher.cs

```diff
@@ -61,64 +62,79 @@ public async Task<string> GetMatchingBranch(string projectKey, IRepository gitRe
         {
             Debug.Assert(sonarQubeService.IsConnected,
                 "Not expecting GetMatchedBranch to be called unless we are in Connected Mode");
+            try
```

Comment:
For future reference, if you haven't changed any of the existing logic when adding a `try-catch` block, please add a review comment to say so.

GitHub doesn't do a great job of showing the changes in cases like this, which makes it harder for reviewers to tell if anything has changed.